### DISCUSSION TITLE
Crazyradio PA Channel Selection

### DISCUSSION THREAD

**Discussion by MouryaT** [bitcraze/discussions#1191] [Q&A] [answered]
Hi, I was wondering if we have to explicitly change crazyradio pa channel ? 
When I change crazyflie config on cfclient to specific channel and address (e.g., crazyflie address 0xE7E7E7E701 and Channel 80 to crazyflie address 0xE7E7E7E705 and Channel 100), the radio detects it automatically while scanning  (radio://0/80/2M/E7E7E7E701 to radio://0/100/2M/E7E7E7E705) and everything works fine.
I did notice set_channel(self, channel) on the [Python Crazyradio Library](https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/functional-areas/crazyradio_lib/) API.

I just began to work on crazyflie hardware. Thanks.

**Comment by ataffanel** [comment, marked as answer]
The connection works with an URI of the form ```radio://<dongle_id>/<channel>/<bitrate>/<address>```. The crazyflie lib will setup the radio according to this URI to connect a Crazyflie. You do not have to touch the Crazyradio API when working with the Crazyflie lib since the lib is handling everything for you.

I hope this clarifies things for you, if not, do not hesitate.

**Comment by MouryaT** [comment]
Yes it does, so cfclient (based on crazyflie lib) automatically scans for crazyflie on the specified address. I noticed, the _select interface_ only displays the URI with  appropriate channel, etc., `radio://<dongle_id>/<channel>/<bitrate>/<address>`  if the crazyflie (using USB for first time) is pre-configured with the specified channel and address. 

![IMG_4795](https://github.com/bitcraze/discussions/assets/15723051/bac1786e-bba1-4580-8cb7-897f86c85454)
![usb_crazyflie_config](https://github.com/bitcraze/discussions/assets/15723051/497ab343-d274-47a3-9fc0-da84b8e91dc0)
![IMG_4798](https://github.com/bitcraze/discussions/assets/15723051/78ce1f7f-34ec-4b62-9a86-a6248a84fb0c)

![IMG_4796](https://github.com/bitcraze/discussions/assets/15723051/2c49bb51-44bb-4b89-b99c-d40f47dd0871)
![IMG_4797](https://github.com/bitcraze/discussions/assets/15723051/08ff33d9-39d5-42c4-b78e-2c3137e3c471)

Thanks.

**Reply by ataffanel** [reply]
What you are seeing in the client is the result of the scan: the client scans for Crazyflie on all the channels and datarate and display the resulting URI in the list. These URI can be used in a Script to connect the Crazyflie (ie. it means that you can connect a Crazyflie over USB from a script if you want).
